src/hotspot/share/opto/compile.cpp

```diff
@@ -3160,6 +3160,32 @@ void Compile::final_graph_reshaping_impl(Node *n, Final_Reshape_Counts& frc, Uni
   }
 }
 
+void Compile::handle_div_mod_op(Node* n, int div_op, int div_mod_op, bool is_unsigned) {
+  if (!UseDivMod) {
+    return;
+  }
+
+  // Check if a%b and a/b both exist
+  Node* d = n->find_similar(div_op);
+  if (!d) {
+    return;
+  }
+
+  BasicType bt = div_op == Op_DivI || div_op == Op_UDivI ? T_INT : T_LONG;
```

Comment:
and add a new function `Op_DivMod(BasicType bt, bool is_unsigned)` similar to `Op_ConIL` that would replace `div_mod_op` here.

Reply:
Added new function `Op_DivModIL(...)` as `Op_DivMod` already exists as an opcode constant and cannot be removed without significant changes to the codebase (although never referenced directly).

Reply:
I don't mind using `Op_DivModIL(...)` but what happens if you remove `Op_DivMod`?

Reply:
I think that's because `DivModNode` has a virtual method `Opcode()` that's defined by `macro`. Removing the `Op_DivMod` line removes the virtual method definition but the declaration is still there. So removing `Op_DivMod` requires removing the `Opcode()` declaration in the `DivModNode` class. Anyway, that can be cleaned up in a separate PR if needed.

Reply:
Thanks for explaining. That is reasonable